src/main/java/duke/main/Parser.java

```diff
@@ -0,0 +1,196 @@
+package duke.main;
+
+import duke.command.AddCommand;
+import duke.command.ByeCommand;
+import duke.command.Command;
+import duke.command.DeleteCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.command.MarkCommand;
+import duke.command.UnmarkCommand;
+import duke.exception.DukeException;
+import duke.exception.InvalidSyntaxException;
+import duke.task.TaskList;
+
+/**
+ * A class handling commands from user input
+ */
+public class Parser {
+
+    /**
+     * A list of valid commands
+     */
+    public enum ValidCommand {
+        BYE, LIST, TODO, DEADLINE, EVENT, MARK, UNMARK, DELETE, UNKNOWN, FIND
+    }
+
+    /**
+     * The current command from user input
+     */
+    private ValidCommand currentCommand;
+
+    /**
+     * Returns Command object based on user input.
+     * @param inp The String array containing the user input
+     * @param taskList The list of tasks under the user
+     * @return Command
+     * @throws DukeException if the format of Command is wrong
+     */
+    public Command parse(String[] inp, TaskList taskList) throws DukeException {
```

Comment:
Very long method, hard to follow. Consider using OOP to break them down into simpler steps?